gadget/layout.go

```diff
@@ -199,13 +198,21 @@ func layoutVolumeStructures(volume *Volume) (structures []LaidOutStructure, byNa
 	for idx := range volume.Structure {
 		ps := LaidOutStructure{
 			VolumeStructure: &volume.Structure[idx],
-			StartOffset:     *volume.Structure[idx].Offset,
 			YamlIndex:       idx,
 		}
 
 		if ps.Name() != "" {
 			byName[ps.Name()] = &ps
 		}
+		// Fill the parts of OnDiskStructure that do not depend on the disk
+		// or on whether we are encrypting or not.
+		// TODO how to fill everything? and about removing startoffset?
```

Comment:
we can't remove StartOffset because when we have partial volume.Structure might not always be set